packages/openapi-generator/src/file-serializer/schema.ts

```diff
@@ -29,7 +29,7 @@ export function serializeSchema(schema: OpenApiSchema): string {
   }
   if (isArraySchema(schema)) {
     const type = serializeSchema(schema.items);
-    return schema.uniqueItems ? `Set<${type}>` : `${type}[]`;
+    return schema.uniqueItems ? `Set<${type}>` : `(${type})[]`;
```

Comment:
[req] I would prefer that we differentiate between a simple array type (e.g. string[]) or array of objects and based on that add the `(..)`

Reply:
And also add tests that cover both cases

Reply:
If the difference is only in `items.properties`, then I think we can add the condition in the array check itself, insetad of another if

Reply:
Yes I thought of doing this yesterday but i wasn't sure what the convention was. I have updated it as requested.